File: Java/Pilha/Stack.java
package Pilha;

public class Stack<T> {
    
    Node<T> topo;
    int length;

    public  Stack(){
        this.topo = null;
        length = 0;
    }

    public boolean isEmpty(){
        if(this.topo == null){
            return true;
        }else{
            return false;
        }
    }

    public void push(T data){
        Node<T> node = new Node<>(data);

        if(isEmpty()){
            this.topo = node;
            length++;
            return;
        }

        node.next = this.topo;
        this.topo = node;
        length++;
        return;

    }

    public T pop(){
        if(isEmpty()) return null;

        T valor = this.topo.data;
        this.topo = this.topo.next;
        length--;
        return valor;
    }

    public T peek(){
        if(isEmpty()) return null;

        T valor = this.topo.data;
        return valor;
    }

}

File: Python/Pilha/Stack.py
from typing import TypeVar,Generic, Optional
from Node import Node

T = TypeVar('T')
class Stack(Generic[T]):

    def __init__(self):
        self.topo:Optional[Node[T]] = None
        self.length = 0    

    def isEmpty(self) -> bool:
        return self.topo is None
    
    def push(self, data:T):
        newNode = Node(data)
        if(self.isEmpty()):
            self.topo = newNode
            self.length += 1
            return
        
        newNode.next = self.topo
        self.topo = newNode
        self.length += 1
        return
    
    def pop(self) -> T:
        if (self.isEmpty()): return None

        nodeData = self.topo.data
        self.topo = self.topo.next
        self.length -= 1
        return nodeData

    def peek(self) -> T:
        if(self.isEmpty()): return None
        nodeData = self.topo.data
        return nodeData 